homeassistant/components/devialet/media_player.py

```diff
@@ -0,0 +1,260 @@
+
+    @property
+    def volume_level(self) -> float | None:
+        """Volume level of the media player (0..1)."""
+        return self._client.volume_level
+
+    @property
+    def is_volume_muted(self) -> bool | None:
+        """Return boolean if volume is currently muted."""
+        return self._client.is_volume_muted
+
+    @property
+    def source_list(self) -> list[str] | None:
+        """Return the list of available input sources."""
+        return self._client.source_list
+
+    @property
+    def sound_mode_list(self) -> list[str] | None:
+        """Return the list of available sound modes."""
+        return sorted(SOUND_MODES)
+
+    @property
+    def media_artist(self) -> str | None:
+        """Artist of current playing media, music track only."""
+        return self._client.media_artist
+
+    @property
+    def media_album_name(self) -> str | None:
+        """Album name of current playing media, music track only."""
+        return self._client.media_album_name
+
+    @property
+    def media_title(self) -> str | None:
+        """Return the current media info."""
+        if not self._client.media_title:
+            return self.source
+
+        return self._client.media_title
+
+    @property
+    def media_image_url(self) -> str | None:
+        """Image url of current playing media."""
+        return self._client.media_image_url
+
+    @property
+    def media_duration(self) -> int | None:
+        """Duration of current playing media in seconds."""
+        return self._client.media_duration
+
+    @property
+    def media_position(self) -> int | None:
+        """Position of current playing media in seconds."""
+        return self._client.current_position
+
+    @property
+    def media_position_updated_at(self) -> datetime.datetime | None:
+        """When was the position of the current playing media valid."""
+        return self._client.position_updated_at
+
+    @property
+    def supported_features(self) -> MediaPlayerEntityFeature:
+        """Flag media player features that are supported."""
+        features = SUPPORT_DEVIALET
+
+        if self._client.source_state is None:
+            return features
+
+        available_options = self._client.available_options
+        if available_options is None:
+            return features
+
+        if "play" in available_options:
+            features = features | MediaPlayerEntityFeature.PLAY
+        if "pause" in available_options:
+            features = (
+                features
+                | MediaPlayerEntityFeature.PAUSE
+                | MediaPlayerEntityFeature.STOP
+            )
+        if "previous" in available_options:
+            features = features | MediaPlayerEntityFeature.PREVIOUS_TRACK
+        if "next" in available_options:
+            features = features | MediaPlayerEntityFeature.NEXT_TRACK
+        if "seek" in available_options:
+            features = features | MediaPlayerEntityFeature.SEEK
```

Comment:
Please turn this into a map instead, and iterate over the keys in `self._client.available_options`

```py
DEVIALET_TO_HA_FEATURE_MAP = {
    "play": MediaPlayerEntityFeature.PLAY,
    ...
}

for option in available_options:
    features |= DEVIALET_TO_HA_FEATURE_MAP.get(option, 0)
```